### PR TITLE
Replace `JwtSecurityToken` with `JsonWebToken`

### PR DESCRIPTION
**Pull request by rubo**
## Changes

- Replaced `JwtSecurityToken` with the new `JsonWebToken` that's up to 30% faster, as Microsoft [claims](https://github.com/dotnet/aspnetcore/issues/49469)
- Made the `JwtAuthentication.Authenticate()` method async because of the underlying handler async API, which in turn caused chained async changes in several places

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [x] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

### CODE REVIEW

**Review by smartprogrammer93 — APPROVED** [review]

File: src/Nethermind/Nethermind.Sockets/IWebSocketsModule.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ namespace Nethermind.Sockets
     public interface IWebSocketsModule
     {
         string Name { get; }
-        ISocketsClient CreateClient(WebSocket webSocket, string client, HttpContext context);
+        Task<ISocketsClient> CreateClient(WebSocket webSocket, string client, HttpContext context);
```
Comment (smartprogrammer93) [inline src/Nethermind/Nethermind.Sockets/IWebSocketsModule.cs:13]: I would use valuetask as we won't need the task later

**Review by smartprogrammer93 — APPROVED** [review]

File: src/Nethermind/Nethermind.Sockets/IWebSocketsModule.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Sockets/IWebSocketsModule.cs:13]: Or no task at all?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Sockets/IWebSocketsModule.cs (hunk 1 above)
Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Sockets/IWebSocketsModule.cs:13]: One of the interface implementations added an await call. Thats why it was added

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Sockets/IWebSocketsModule.cs:13]: ah and it is the main one

**Review by benaadams — APPROVED** [review]

**Pull request merged**